Contributing guidelines

ayushanand18 opened the pull request:
## Overview
This PR aims to add Contributor Guidelines on CONTRIBUTING.md to the pyobis project.
+ It aims to resolve #19 

## Notes for Reviewers
Please review the Contributor Guidelines and highlight changes if I need to add some more points here.

## Next Steps
I am now working on creating a PR for the `usage_guide.ipynb` file post this review.

7yl4r commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,48 @@
+2. If you've added code that should be tested, add tests.
+3. If you've changed APIs, update the documentation.
+4. Ensure the test suite passes.
+5. Make sure your code lints.
```
we can drop 4 and 5 here. We'll handle that using GH actions after the pull request is submitted.

ayushanand18 commented:
I have updated the contributing guide as per the changes suggested. I have also added some sections:
+ `why contribute to the open source?`
+ `pyobis, and its aim`
+ `additional suggestions`
+ `further reading`
